### ISSUE TITLE
False positive for PEP 604 `None | t` when used as value

### ISSUE THREAD

**Issue by NiklasRosenstein** [python/mypy#12368]
**Bug Report**

Using `None | t` instead of `t | None` prompts an error when used as a value (not a type hint).

https://peps.python.org/pep-0604/#proposal

**To Reproduce**

```py
a: None | int
b = None | int  # error: Unsupported left operand type for | ("None")  [operator]
```

**Expected Behavior**

Mypy does not complain about the second line.

**Your Environment**

- Mypy version used: 90.940
- Mypy command-line flags: 
- Mypy configuration options from `mypy.ini` (and other config files):
    ```toml
    pretty = true
    namespace_packages = true
    warn_redundant_casts = true
    warn_no_return = true
    warn_unreachable = true
    show_error_context = true
    show_error_codes = true
    check_untyped_defs = true
    ```
- Python version used: 3.10.2
- Operating system and version: Ubuntu 20.04 WSL

**Comment by jhance** [comment]
I assume this means that `isinstance(5, int | None)` fails, which seems like a major bug.

**Comment by NiklasRosenstein** [comment]
@jhance  Your assumption is correct.
